pkg/bpf/bpf.go

```diff
@@ -305,6 +314,10 @@ func OpenOrCreateMap(path string, mapType int, keySize, valueSize, maxEntries ui
 		if err != nil {
 			return 0, isNewMap, err
 		}
+
+		// Need to close this fd since the next ObjGet
+		// returns a different fd.
+		ObjClose(fd)
```

Comment:
I'm not sure this one is correct: when above ObjPin() fails, we need to close the prior created fd in ObjPin()'s error path. And I don't think it makes much sense to perform a ObjClose() just to fall-through in order to later do a ObjGet(), seems unnecessary to me. Why not just return the fd at that point, since we already have one?